ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto

```diff
@@ -180,20 +190,25 @@ message CreateUserResponse {
     User user = 1;
 }
 
-// Required authorization: ``HasRight(ParticipantAdmin) OR IsAuthenticatedUser(user_id)``
+// Required authorization: ``HasRight(ParticipantAdmin) OR HasRight(IdentityProviderAdmin) OR IsAuthenticatedUser(user_id)``
```

Comment:
This authorization rule looks fishy. From @mziolekda 's description today, I'd expect that one of the admin rights is restricted to the requestor having the right to adminstrate users of a specific identity provider.

Otherwise why have `ParticipantAdmin` and `IdentityProviderAdmin`?

Reply:
I am not sure I understand fully, but if you are saying that the rule should say 
```
HasRight(ParticipantAdmin) OR IsAuthenticatedIdentityProviderAdmin(identity_provider_id) OR IsAuthenticatedUser(user_id)
```
then I agree

Reply:
What's the relation between a `identity_provider_id` and a `user_id`? Is an `identity_provider_id` a `user_id`?

Reply:
no, think of `identity_provider_id` as an alias of the idp. while the `issuer` and `iss` claims are both urls and can be modified, this alias is going to be stable.

Reply:
Notes from our discussion:

```
HasRight(ParticipantAdmin) OR            // participant admins can administer all users
(HasRight(IdentityProviderAdmin) AND     
  authenticatedUser.identity_provider == request.identity_provider_id
) OR                                     // users that are identity provider admins can
                                         // administrate all users of the same identity provider
(authenticated_user.user_id == request.user_id) OR  // users can fetch their own users records
request.user_id == ""

```

Reply:
I suggest following change in the preamble of this file
```
// (1) ``HasRight(r)`` denoting whether the authenticated user has right ``r`` and
// (2) ``IsAuthenticatedUser(uid)`` denoting whether ``uid`` is equal to the id of the authenticated user.
// (3) ``IsAuthenticatedIdentityProviderAdmin(idp)`` denoting whether ``idp`` is equal to the ``identity_provider_id``
// of the authenticated user.
```
Then throughout, we can use a shorthand notation to denote the authorization rules like so:
```
HasRight(ParticipantAdmin) OR 
IsAuthenticatedIdentityProviderAdmin(identity_provider_id) OR 
IsAuthenticatedUser(user_id)
```
The rules for dealing with empty `user_id` are orthogonal to the authorization and I would explain them separately, as effectively empty `user_id` means implying it is equal to authenticated user i.e. not empty.

Reply:
`// (2) ``IsAuthenticatedUser(uid)`` denoting whether ``uid`` is equal to the id of the authenticated user.` 
Here you removed `is the empty string`, I am wondering how do we denote that we support an empty user_id given the user is requesting the information about himself?